anndata/utils.py

```diff
@@ -254,3 +267,18 @@ def func(*_, **__):
             raise error
 
     return func
+
+
+@singledispatch
+def dim_len(x, dim):
+    return x.shape[dim]
```

Comment:
```python
awkward.size(array, dim)
```

does exactly that for both numpy and awkward arrays. 
If the requested dimension has variable length, it raises a `ValueError`